lib/LanguageServerCompletion/Handler/CompletionHandler.php

```diff
@@ -166,20 +175,31 @@ private function textEdit(Suggestion $suggestion, TextDocumentItem $textDocument
         );
     }
 
-    private function command(string $uri, ByteOffset $offset, Suggestion $suggestion): ?Command
-    {
-        if (!$suggestion->nameImport()) {
-            return null;
+    private function importClassOrFunctionName(
+        Suggestion $suggestion,
+        CompletionParams $params
+    ): ImportNameResult {
+        $suggestionNameImport = $suggestion->nameImport();
+
+        if (!$suggestionNameImport) {
+            return ImportNameResult::createEmptyResult();
         }
 
-        if (!in_array($suggestion->type(), [ 'class', 'function'])) {
-            return null;
+        $suggestionType = $suggestion->type();
+
+        if (!in_array($suggestionType, [ 'class', 'function'])) {
+            return ImportNameResult::createEmptyResult();
         }
 
-        return new Command(
-            'Import class',
-            ImportNameCommand::NAME,
-            [$uri, $offset->toInt(), $suggestion->type(), $suggestion->nameImport()]
+        $doc = $this->workspace->get($params->textDocument->uri);
+        // todo: avoid conversions?
+        $offset = PositionConverter::positionToByteOffset($params->position, $doc->text);
```

Comment:
I found several places where positions are converted. After all I found three options: Integer, ByteOffset and Position. Maybe not a task for this PR but in general it maybe would make sense to replace all of these by an unified Position providing line/col and the byte offset. That could save many conversion calls.

Reply:
In general Phpactor is not coupled to LSP (for historical reasons) so this is where we cross domains. Phpactor has it's own value objects in `phpactor/text-document`: `LineCol` for example translates to LSP `Position`.

On the positive side the Phpactor V/Os are much more useful to work with, as they can provide utility methods, whereas the LSP protocol classes are basically DTOs.